ci/test/00_setup_env_mac_native.sh

```diff
@@ -11,7 +11,7 @@ export HOST=x86_64-apple-darwin
 # Therefore, `--break-system-packages` is needed.
 export PIP_PACKAGES="--break-system-packages zmq"
 export GOAL="install"
-export BITCOIN_CONFIG="--with-gui --with-miniupnpc --with-natpmp --enable-reduce-exports"
```

Comment:
When usages of `--with-natpmp` are removed from the CI in commit 40908bbb88 `ci: Remove natpmp build option` then this option is already nonexistent, meaning that in the intermediate commits the CI is broken. Maybe reorder the commits so the CI one is before the `--with-natpmp` removal (a749ab8cfe `build: Drop libnatpmp from build system`) or squash them into one commit.